### PR TITLE
Additional context and clarity in processing doc

### PR DESCRIPTION
**Pull request by philipkiely-baseten**

### CODE REVIEW

File: docs/develop/processing.md [hunk 1]
```diff
@@ -1,17 +1,59 @@
...
+An efficiently serialized model is just the model and its direct dependencies. No custom logic or extra frameworks.
 
-One feature that makes Truss different from model serialization frameworks is bundled pre- and post-processing. Most models intended for use in production systems will want to have these functions as part of the Truss.
+This makes deserialization fast, but may make using the model inconvenient.
 
-Add pre-processing and post-processing functions to your Truss by editing `model/model.py` to include the necessary functions.
+Truss bundles your serialized model with pre- and post-processing functions. Use them to:
 
-## Pre-processing
+* Format model inputs and outputs, especially when the model needs something like a Numpy array that isn't JSON-serializable
```
Comment (bolasim) [inline docs/develop/processing.md:9]: i think technically we handle most numpy objects for serialization, but maybe mention xgboost.DMatrix instead which is a more common issue

File: docs/develop/processing.md [hunk 2]
```diff
@@ -32,17 +73,18 @@ def preprocess(self, model_input: Any) -> Any:
...
 Similarly, the output of a ML model can be messy or cryptic. Demystify the model results for your end user or format a web-friendly response in the post-processing function.
 
+**Example: ResNet post-processing**
+
 Here is a post-processing function from the same [TensorFlow example](../create/tensorflow.md) that returns clean labels.
 
 ```python
 from scipy.special import softmax
 
 def postprocess(self, model_output: Any, k=5) -> Any:
```
Comment (bolasim) [inline docs/develop/processing.md:87]: this has the wrong model signature. You shouldn't have `k` like that. it works but isn't a good example that people should follow

File: docs/develop/processing.md [hunk 3]
```diff
@@ -54,3 +96,14 @@ def postprocess(self, model_output: Any, k=5) -> Any:
...
+## Using processing functions
+
+Once you've modified your processing functions, you can test them by reloading the Truss and invoking the model.
+
+```python
+tr = truss.load("my-truss")
+tr.predict("MODEL_INPUT")
+```
+
+&nbsp;
```
Comment (bolasim) [inline docs/develop/processing.md:109]: nit: remove

**Review by bolasim — APPROVED** [review]
